src/components/SideDrawer/__tests__/SideDrawer.test.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { GlobalContext } from '@/contexts/ContextProvider';
+import { SideDrawer } from '../SideDrawer';
+import { sideMenu } from '@/config/menu';
+import { useRouter } from 'next/router';
+
+jest.mock('next/router', () => ({
+  useRouter: jest.fn(),
+}));
+
+const mockedUseRouter: any = useRouter;
+
+const mockGlobalContextValue: any = {
+  Permissions: { state: ['permission1', 'permission2'] },
+  CurrentOrg: { state: { is_demo: true } },
+};
+
+describe('SideDrawer', () => {
+  const setOpenMenu = jest.fn();
+  const push = jest.fn();
+
+  beforeEach(() => {
+    mockedUseRouter.mockReturnValue({
+      pathname: '/',
+      push,
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  expect(sideMenu).toBeDefined();
+  expect(sideMenu.length).toBeGreaterThan(0);
+
+  beforeEach(() => {
```

Comment:
Use a single beforeEach function for a test file

Reply:
ok. I ll check that. Thanks.

Reply:
Added single beforeEach done.